narwhals/stable/v1/__init__.py

```diff
@@ -865,14 +891,54 @@ def func(df):
         allow_series: Whether to allow series (default is only dataframe / lazyframe).
     """
 
-    return nw_narwhalify(
-        func=func,
-        strict=strict,
-        eager_only=eager_only,
-        eager_or_interchange_only=eager_or_interchange_only,
-        series_only=series_only,
-        allow_series=allow_series,
-    )
+    def decorator(func: Callable[..., Any]) -> Callable[..., Any]:
+        @wraps(func)
```

Comment:
this reverts https://github.com/narwhals-dev/narwhals/pull/799 , as then the stable `narwhalify` would return non-stable objects

cc @FBruzzesi , just FYI (no blame 🤗 )